utils/emmylua/wesnoth.lua

```diff
@@ -0,0 +1,272 @@
+---| "'local_choice'"
+---| "'preload'"
+
+---Holds information about the current game state
+---@class current
+---@field side integer Number of the side that currently has control
+---@field turn integer Number of the current turn
+---@field synced_state synced_state
+---@field user_can_invoke_commands boolean
+---@field map terrain_map
+---@field schedule schedule
+---@field event_context event_context
+wesnoth.current = {}
+
+---Holds global game configuration options
+---@class game_config
+---@field base_income integer
+---@field village_income integer
+---@field village_support integer
+---@field poison_amount integer
+---@field rest_heal_amount integer
+---@field recall_cost integer
+---@field combat_experience integer
+---@field kill_experience integer
+---@field global_traits table<string, WMLTable>
+---@field do_healing boolean
+---@field theme string
+---@field debug boolean
+---@field debug_lua boolean
+---@field mp_debug boolean
+---@field strict_lua boolean
+wesnoth.game_config = {}
+
+---@type table<string, unit_race>
+wesnoth.races = {}
+
+---Holds information about the current scenario
+---@class scenario
+---@field turns integer
+---@field next string|nil
+---@field id string
+---@field defeat_music string[]
+---@field victory_music string[]
+---@field show_credits boolean
+---@field end_text tstring
+---@field end_text_duration number
+---@field difficulty string
+---@field type "'scenario'"|"'test'"|"'multiplayer'"
+---@field era WMLTable|nil
+---@field campaign WMLTable|nil
+---@field resources WMLTable[]
+---@field modifications WMLTable[]
+---@field mp_settings mp_settings
+wesnoth.scenario = {}
+
+---@class mp_settings
+---@field game_name string
+---@field hash string
+---@field experience_modifier integer
+---@field mp_countdown boolean
+---@field mp_countdown_init_time integer
+---@field mp_countdown_turn_bonus integer
+---@field mp_countdown_reservoir_bonus integer
+---@field mp_countdown_action_bonus integer
+---@field mp_village_gold integer
+---@field mp_village_support integer
+---@field mp_fog boolean
+---@field mp_shroud boolean
+---@field mp_use_map_settings boolean
+---@field mp_random_start_time boolean
+---@field allow_observers boolean
+---@field private_replay boolean
+---@field shuffle_sides boolean
+---@field random_faction_mode "'Independent'"|"'No Mirror'"|"'No Ally Mirror'"
+---@field options WMLTable
+---@field savegame 'false'|"'midgame'"|"'scenario_start'"
+---@field side_players table<string, string>
+---@field addons mp_addon[]
+---@class mp_addon
+---@field id string
+---@field name string
+---@field required boolean
+---@field min_version? string
+---@field version? string
+---@field content {id:string, name:string, type:string}
+
+---Information on a specific terrain type
+---@class terrain_info
+---@field id string
+---@field name tstring
+---@field editor_name tstring
+---@field description tstring
+---@field icon string
+---@field editor_image string
+---@field light integer
```

Comment:
The C++ classes have minimum / maximum light too, but this might be missing from the LuaAPI.

Reply:
Yeah, looks like those are not exposed in the LuaAPI. Think we should add them? (Not in this PR obviously.)

Reply:
OK. As for adding them, maybe review the C++ and the WML definitions and see if we want to change them first. That effort would fit in with the remaining documentation in #4417, seeing if there's a easier way to describe it than the one currently on the TerrainWML page.